include/manif/impl/se_2_3/SE_2_3Tangent_base.h

```diff
@@ -0,0 +1,394 @@
+
+#include "manif/impl/se_2_3/SE_2_3_properties.h"
+#include "manif/impl/tangent_base.h"
+#include "manif/impl/so3/SO3Tangent_map.h"
+#include "manif/impl/se3/SE3Tangent_map.h"
+
+namespace manif {
+
+//
+// Tangent
+//
+
+/**
+ * @brief The base class of the SE_2_3 tangent.
+ */
+template <typename _Derived>
+struct SE_2_3TangentBase : TangentBase<_Derived>
+{
+private:
+
+  using Base = TangentBase<_Derived>;
+  using Type = SE_2_3TangentBase<_Derived>;
+
+public:
+
+  MANIF_TANGENT_TYPEDEF
+  MANIF_INHERIT_TANGENT_OPERATOR
+
+  using BlockV = typename DataType::template FixedSegmentReturnType<3>::Type;
+  using BlockW = typename DataType::template FixedSegmentReturnType<3>::Type;
+  using BlockA = typename DataType::template FixedSegmentReturnType<3>::Type;
+  using ConstBlockV = typename DataType::template ConstFixedSegmentReturnType<3>::Type;
+  using ConstBlockW = typename DataType::template ConstFixedSegmentReturnType<3>::Type;
+  using ConstBlockA = typename DataType::template ConstFixedSegmentReturnType<3>::Type;
+
+  using Base::data;
+  using Base::coeffs;
+
+  SE_2_3TangentBase()  = default;
+  ~SE_2_3TangentBase() = default;
+
+  // Tangent common API
+
+  /**
+   * @brief Hat operator of SE_2_3.
+   * @return An element of the Lie algebra se_2_3.
+   * @note See Eq. (169).
+   */
+  LieAlg hat() const;
+
+  /**
+   * @brief Get the SE_2_3 element.
+   * @param[out] -optional- J_m_t Jacobian of the SE_2_3 element wrt this.
+   * @return The SE_2_3 element.
+   * @note This is the exp() map with the argument in vector form.
+   */
+  LieGroup exp(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief This function is deprecated.
+   * Please considere using
+   * @ref exp instead.
+   */
+  MANIF_DEPRECATED
+  LieGroup retract(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief Get the right Jacobian of SE_2_3.
+   */
+  Jacobian rjac() const;
+
+  /**
+   * @brief Get the left Jacobian of SE_2_3.
+   */
+  Jacobian ljac() const;
+
+  /**
+   * @brief Get the small adjoint matrix ad() of SE_2_3
+   * that maps isomorphic tangent vectors of SE_2_3
+   * @return
+   */
+  Jacobian smallAdj() const;
+
+  // SE_2_3Tangent specific API
+
+  //! @brief Get the linear velocity part.
+  BlockV v();
+  const ConstBlockV v() const;
+
+  //! @brief Get the angular part.
+  BlockW w();
+  const ConstBlockW w() const;
+
+  //! @brief Get the linear acceleration part
+  BlockA a();
+  const ConstBlockA a() const;
+
+public: /// @todo make protected
+
+  const Eigen::Map<const SO3Tangent<Scalar>> asSO3() const
+  {
+    return Eigen::Map<const SO3Tangent<Scalar>>(coeffs().data()+3);
+  }
+
+  Eigen::Map<SO3Tangent<Scalar>> asSO3()
+  {
+    return Eigen::Map<SO3Tangent<Scalar>>(coeffs.data()+3);
+  }
+
+};
+
+template <typename _Derived>
+typename SE_2_3TangentBase<_Derived>::LieGroup
+SE_2_3TangentBase<_Derived>::exp(OptJacobianRef J_m_t) const
+{
+  using std::sqrt;
+  using std::cos;
+  using std::sin;
```

Comment:
It looks like you don't need those, please remove them.

Reply:
Fixed in https://github.com/artivis/manif/pull/154/commits/810862f30d3e97e9de9dbd54e1a7458146ae91f5